tests/unit/query_model/test_nodes.py

```diff
@@ -352,6 +353,29 @@ def test_cannot_aggregate_a_one_row_per_patient_series():
         AggregateByPatient.Max(first_date)
 
 
+def test_can_construct_dataset():
+    events = SelectTable("events", EVENTS_SCHEMA)
+    dates = SelectColumn(events, "date")
+    assert Dataset(
+        population=AggregateByPatient.Exists(events),
+        variables={"max_date": AggregateByPatient.Max(dates)},
+    )
+
+
+def test_can_construct_empty_dataset():
+    assert Dataset(population=Value(False), variables={})
```

Comment:
No, nothing significant about `Value(False)` here other than being the simplest possible valid population definition. This test only exists because one of my previous attempts at the domain validation code choked when given a empty variables dict. I think it's worth keeping but I'll add a comment.